### PR TITLE
Adding up and down vote functionality to reviews

### PR DESCRIPTION
**Pull request by omris-elda**

### CODE REVIEW

File: src/components/IndvReview/IndvReview.jsx [hunk 1]
```diff
@@ -33,11 +30,13 @@ export default function IndvReview() {
...
-            </p>
-                <p className="indv-review-votes">Votes: {review.votes}</p>
+                </p>
+                <PatchReviewVote review_id={review.review_id} votes={review.votes} />
+                {/* <p className="indv-review-votes">Votes: {review.votes}</p> */}
+
 
-                <h3 className="comment-title">Comments:</h3>
-            <Comments comment_id={review_id} />
+                {/* <h3 className="comment-title">Comments:</h3> */}
```
Comment (mallionaire) [inline src/components/IndvReview/IndvReview.jsx]: Should not be any commented out code in your components

File: src/components/PatchReviewVote/PatchReviewVote.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            return null;
+        }
+        setVote((vote) => vote + 1);
+        PatchReviewVotes(review_id, 1);
+    }
+
+    const HandleDownvote = (e) => {
+      e.preventDefault();
+      if (!review_id) {
+        console.log("You shouldn't be seeing this, something has gone horribly wrong!");
```
Comment (mallionaire) [inline src/components/PatchReviewVote/PatchReviewVote.jsx]: We will need to handle errors slightly differently, errors should not be logged to the console.

File: src/components/ReviewCard/ReviewCard.jsx [hunk 3]
```diff
@@ -18,8 +18,10 @@ export default function ReviewCard({ review }) {
           <img src={review.review_img_url} alt={`For review ${review.title}`} className="review-image" />
           <div className="review-inner">
             <p className="review-text">{review.review_body}</p>
+            
             {/* functionality for upvoting? */}
```
Comment (mallionaire) [inline src/components/ReviewCard/ReviewCard.jsx]: No comments in components please

**Review by mallionaire — CHANGES_REQUESTED** [review]
Nice job on the upvote/downvote. 
Here are some things for you to think about, as well as some comments addressing some specific thoughts. 

There is a build up of React errors being logged to the console. You must address these immediately. 
I'm not too keen on the name of your PatchReviewVotes component. This name describes what a small part of the functionality of the component does, rather than what it displays. So think carefully about your component names going forwards. `Votes` might be more accurate (and could potentially make this more reusable in the future for comment votes 🤔 )

No console.logs or commented out code should be included on pull requests.

File: src/components/PatchReviewVote/PatchReviewVote.jsx (hunk 2 above)
Comment (omris-elda) [reply src/components/PatchReviewVote/PatchReviewVote.jsx]: Removed this as this shouldn't be possible, was only there for while I was trying to fix it.

File: src/components/IndvReview/IndvReview.jsx (hunk 1 above)
Comment (omris-elda) [reply src/components/IndvReview/IndvReview.jsx]: This was commented out so I wasn't overstepping the bounds of what should have been done for this task, have uncommented it now and re-added comment functionality.

**Comment by k-huggs** [comment]
When you downvote below 0, you get a 400 error from your backend.

When you click on an article the votes value is different to the one displayed on the reviews section and they are all the same on the individual article. 

Your votes component is a couple of forms, but they have the same ID's, an id is supposed to be unique. You can also do this in two buttons with an onClick function, rather than the forms, then you only need one HandleVote function to patch the votes up or down

**Comment by k-huggs** [comment]
I'm not seeing the read more button to go to the individual articles? 

camel case - PatchReviewVotes function 

Voting function is fine, but I guess you can debate whether you want votes to go below 0 or not

**Comment by omris-elda** [comment]
I've removed the read more functionality, may revisit it later if I have time but didn't want to waste loads of time playing around with it for now as it'd mean changing where and how the logic happens. I don't want votes to go below 0 as I think that'd be a back end thing and I don't want to mess around with that right now. PatchReviewVotes function currently matches the rest of the query functions, should they all be camel case?

**Comment by omris-elda** [comment]
Changed all query functions to be camel case

**Pull request merged**